### DISCUSSION TITLE
[MONGOOSE] DeprecationWarning

### DISCUSSION THREAD

**Discussion by andy-schwedler** [neuefische/questions#447] [Web] [answered]
Hi there, 

I receive a warning in my terminal:

```
(node:24181) [MONGOOSE] DeprecationWarning: Mongoose: the `strictQuery` option will be switched back to `false` by default in Mongoose 7. Use `mongoose.set('strictQuery', false);` if you want to prepare for this change. Or use `mongoose.set('strictQuery', true);` to suppress this warning.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

From what I was able to google this far, this is connected to npm version 7.x.x. Luckily, it does not affect rendering data that is stored in the database. 

I do not know if there´s any disadvantage in setting `strictQuery` true.

Does anyone have the same issue? or knows how to fix this properly?

**Comment by pixelass** [comment, marked as answer]
> back to `false` by default 

This just means that the default value will be changed back (to something that was recently changed). 

The text further states that you can "prepare" for the change (therefore it's a deprecation warning)

You can in the same way prepare (in keeping the current behavior) by setting the value to true.

Conclusion: This is a warning, that proposes an explicit setting in either direction to prevent future conflicts. You can safely ignore it and fix issues if they occur after an update of mongoose.

**Reply by andy-schwedler** [reply]
Thank you @pixelass, that was helpful!
